Cal/ctrl batch flow

CalNightingale opened the pull request:
Refactors the pinch valve field patch

CalNightingale committed:
Initial Commit

CalNightingale committed:
Added Log functions to all events

CalNightingale committed:
Added additional log statements

CalNightingale committed:
Lowered level of temp_read() logs

CalNightingale committed:
Initial commit

CalNightingale committed:
Added comments

CalNightingale committed:
Refactored; replaced ISITUP with pinchValve vars

mlp6 committed:
update README

mlp6 committed:
mlp6/fix-docker-225 (#228)

* change Docker base image from Ubuntu 14.04 -> Debian Jessie (consistent with co-lab VM) and bring along associated NodeJS (6.x) and MongoDB packages

* commit a placeholder twilio_auth.js file that is necessary for the node server to run (fixes #225)

CalNightingale committed:
Removed unnecessary nested if statement

CalNightingale committed:
Refactored bucket.tip

CalNightingale committed:
Replaced WAIT_TIME with a pinch valve attribute

CalNightingale committed:
Merge branch 'master' into cal/log

CalNightingale committed:
Added logs for ignition code

CalNightingale committed:
Added log statements

mlp6 commented on software/ADPL_electron/ADPL_electron.ino:
```diff
@@ -124,27 +128,32 @@ void loop() {
+        if(pinchValve.up){
+            Log.info("Pinch valve is up. Lowering...");
+            // if the pinch valve is up
+            pinchValve.down = true;
```
@CalNightingale do we need both `pinchValve.up` and `pingValve.down` (i.e., is one always opposite the other?  if so, then no need for both... but if both could be `false`, or something like that, then maybe both are needed).

CalNightingale replied:
We don't need both; I have removed `pinchValve.down` and replaced it with `!pinchValve.up` in all locations and wherever `pinchValve.down` was set, `pinchValve.up` is now set to the opposite value

mlp6 commented on software/ADPL_electron/ADPL_electron.ino:
```diff
@@ -72,6 +73,9 @@ void setup() {
     attachInterrupt(RESET, res_pushed, FALLING);
+
+    //initialize pinch valve time attribute
+    pinchValve.wait_time = 0;
```
@CalNightingale what does `wait_time` actually represent?  maybe this could be renamed to be a bit more explicit in its use.

CalNightingale replied:
It represents the last time the pinch valve was moved. I have changed it's name to lastTime, which is consistent with the syntax of the bucket (which has a similar variable)

mlp6 commented on software/ADPL_electron/ADPL_electron.ino:
```diff
@@ -52,6 +52,7 @@ PinchValve pinchValve(DIR, STEP, SLEEP, UP, DOWN, RESET);
 #define UNCLOG_RESOLUTION 4.0 // mm of movment
 #define MAX_POSITION 5.0 // in mm
```
@CalNightingale do we still need `MAX_POSITION` and `MIN_POSITION`?

CalNightingale replied:
We don't! I have removed them.

mlp6 reviewed (changes requested):
@CalNightingale a few comments for your consideration

CalNightingale committed:
Removed unused position variables

CalNightingale committed:
Replaced wait_time with lastTime

This makes it more clear what the purpose of the variable is, and is consistent with the syntax of the bucket object

CalNightingale committed:
Removed redundant pinchValve.down variable

Replaced all instances of said variable with "!pinchValve.up", and wherever down was set, up is now set to the opposite

CalNightingale commented:
Before committing this, I think it would be a good idea to test it on a working setup, since we currently have no idea whether it works as expected. Is there such a setup somewhere around here that I could test this code on?

CalNightingale committed:
Added newline character at end of file

CalNightingale committed:
Fixed bracket syntax

CalNightingale committed:
Removed bad log statement

mlp6 commented:
They will have a setup that you can test on.  Both @aforbis-stokes and Graham will be back in the country next week and can help you with this.  You can commit things to this branch... just no merging until tested.

CalNightingale committed:
Replaced inverse multiplication with division

CalNightingale committed:
Changed res_pushed() function based on field code

CalNightingale committed:
Moved button pins

They are now adjacent to one another, and out of the way of the SD card

CalNightingale committed:
Merge branch 'cal/ctrl-batch-flow' into cal/log

CalNightingale committed:
Merge pull request #230 from mlp6/cal/log

Cal/log_AND_ctrl_flow

mlp6 commented:
@CalNightingale has this been adequately tested, or are you planning to do more?

CalNightingale committed:
Removed unused pinch valve vars

CalNightingale committed:
Fixed up button text

CalNightingale committed:
Fixed down button indentation

CalNightingale committed:
commented out .up calls

CalNightingale committed:
Misc changes to get motor working

CalNightingale committed:
Changed pin mapping

mlp6 commented:
This branch is outdated and certain changes were manually cherry picked onto the `cal/fatlibMigration` branch.
